File: USACO/Bronze/USACO 2021 February Contest Bronze/Clockwise_Fence/Clockwise_Fence.py
def solve_line(line):
    r = 0
    l = 0

    for i in range(len(line)-1):
        temp = line[i:i+2]
        if ((temp == "NE") or (temp == "ES") or (temp == "SW") or (temp =="WN")):
            r += 1
        elif ((temp == "EN") or (temp == "SE") or (temp == "WS") or (temp == "NW")):
            l += 1
    
    if r > l:
        return 'CW'
    else:
        return 'CCW'

            
if __name__ == "__main__":
    N = int(input().strip())
    ans_l = []

    for i in range(N):
        line = input().strip()
        ans = solve_line(line)
        ans_l.append(ans)
    
    for i in ans_l:
        print(i)

File: USACO/Bronze/USACO 2021 February Contest Bronze/Clockwise_Fence/clockwise_fence.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

string solve_line(string line)
{   
    int r=0, l=0;
    string temp;

    for(int i=0; i<line.length()-1; i++)
    {   
        temp=line.substr(i,2);
        if((temp == "NE") || (temp == "ES") || (temp == "SW") || (temp =="WN"))
        {
            r++;
        }
        else if((temp == "EN") || (temp == "SE") || (temp == "WS") || (temp == "NW"))
        {
            l++;
        }
    }

    // cout << "r: " << r << endl;
    // cout << "l: " << l << endl;
    if(r>l)
    {
        return "CW";
    }
    else
    {
        return "CCW";
    }
}

int main()
{
    int N;
    cin >> N;

    vector<string> ans(N);
    string line;

    
    for(int i=0; i<N; i++)
    {   
        cin >> line;
        ans[i] = solve_line(line);
    }
    
    for(int i=0; i<N; i++)
    {
        cout << ans[i] << endl;
    }
    return 0; 
}

